CMakeLists.txt

```diff
@@ -386,18 +386,21 @@ if(ZLIB_FOUND)
 endif()
 
 ############################################################################
-# Check for Qt5
+# Check for Qt
 ############################################################################
 
-find_package(Qt5Widgets 5.15)
-if(Qt5Widgets_FOUND)
-	set(CMAKE_STATUS_QT_VERSION "${Qt5Widgets_VERSION_STRING}")
-	set(qt5_kvirc_modules)
+# first check if Qt6 or Qt5 has to be used
+find_package(QT NAMES Qt6 Qt5 REQUIRED COMPONENTS Core)
+# second, detect available modules for the specific Qt version
```

Comment:
Why not do both components in the same find_package? `COMPONENTS Core Widgets`

Reply:
The first call is only used to determine if Qt5 or Qt6 will be used, and define `${QT_VERSION_MAJOR}` accordingly.
All the other calls force the specific Qt version using `find_package(Qt${QT_VERSION_MAJOR} ..)`.

This way of support both Qt5 and Qt6 without mixing them is documented here: https://doc.qt.io/qt-6/cmake-qt5-and-qt6-compatibility.html